.github/workflows/release.yml

```diff
@@ -0,0 +1,146 @@
+        working-directory: plugins/maven/graphql-java-codegen-maven-plugin
+        run: mvn install
+
+      - name: Build sbt plugin
+        working-directory: plugins/sbt/graphql-java-codegen-sbt-plugin
+        run: sbt compile publishLocal --debug
+
+      - name: Build sbt test
+        working-directory: plugins/sbt/graphql-java-codegen-sbt-plugin
+        run: sbt scripted
+
+      - name: Build gradle example-server --warning-mode all
+        run: ./gradlew -p plugins/gradle/example-server test --warning-mode all
+
+      - name: Build gradle example-client
+        run: ./gradlew -p plugins/gradle/example-client test --warning-mode all
+
+      - name: Build gradle example-client-kotlin
+        run: ./gradlew -p plugins/gradle/example-client-kotlin build --warning-mode all
+
+      - name: Build maven example-server
+        working-directory: plugins/maven/example-server
+        run: mvn package
+
+      - name: Build maven example-client
+        working-directory: plugins/maven/example-client
+        run: mvn package
+
+      - name: Install gpg secret key
+        run: cat <(echo -e "${{ secrets.OSSRH_GPG_SECRET_KEY }}") | gpg --batch --import
+
+      #
+      # RELEASE STEPS
+      #
+
+      - name: Release library
+        run: |
+          ./gradlew publish \
+              -Dorg.gradle.internal.publish.checksums.insecure=true \
+              -Pgradle.publish.key=${{ secrets.GRADLE_PUBLISH_KEY}} \
+              -Pgradle.publish.secret=${{ secrets.GRADLE_PUBLISH_SECRET }}
+        env:
+          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+          SIGNING_KEY: ${{ secrets.OSSRH_GPG_SECRET_KEY }}
+          SIGNING_PASSWORD: ${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }}
+          RELEASE_VERSION: ${{ github.event.inputs.release_version }}
+
+      - name: Release Gradle plugin
+        run: |
+          ./gradlew \
+              -p plugins/gradle/graphql-java-codegen-gradle-plugin \
+              publishPlugins \
+              -Dorg.gradle.internal.publish.checksums.insecure=true \
+              -Pgradle.publish.key=${{ secrets.GRADLE_PUBLISH_KEY}} \
+              -Pgradle.publish.secret=${{ secrets.GRADLE_PUBLISH_SECRET }}
+        env:
+          SIGNING_KEY: ${{ secrets.OSSRH_GPG_SECRET_KEY }}
+          SIGNING_PASSWORD: ${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }}
+          RELEASE_VERSION: ${{ github.event.inputs.release_version }}
+
+      - name: Release Maven plugin
+        working-directory: plugins/maven/graphql-java-codegen-maven-plugin
+        run: |
+          mvn \
+            --no-transfer-progress \
+            --batch-mode \
+            -Dgpg.passphrase=${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }} \
+            release:clean release:prepare release:perform
+
+      - name: Release SBT plugin
+        working-directory: plugins/sbt/graphql-java-codegen-sbt-plugin
+        run: sbt release with-defaults default-tag-exists-answer k
```

Comment:
This will use the default version, such as (3.1.1 - > 3.1.2-SNAPSHOT), But other plugins use `${{ github.event.inputs.release_version }}`, This may need to change.

Like this, We can use args  `release-version 1.0.99 next-version 1.2.0-SNAPSHOT`
![image](https://user-images.githubusercontent.com/29496873/109916457-9fccec80-7cee-11eb-99b2-174bcd2808e0.png)

Reply:
@jxnu-liguobin this workflow will always be executed from the master branch which will not have any SHANPSHOT versions.

Reply:
> @jxnu-liguobin this workflow will always be executed from the master branch which will not have any SHANPSHOT versions.

Will the next version be updated after the release?

Reply:
The idea is the following:
In **master branch** (from which the release will be made) - we will keep the original release version and do not increment to the next SNAPSHOT version.
In **develop branch** (after release will be made) - we will increment to the next SNAPSHOT version.

Reply:
very good! 👍